src/snarky.d.ts

```diff
@@ -51,489 +53,806 @@ declare interface ProvablePure<T> extends Provable<T> {
    */
   toBigInt(): bigint;
-  /**
-   * Serialize this instance of a {@link Field} to a JSON string.
-   * This operation does NOT affect the circuit and can't be used to prove anything about the string representation of the Field.
-   */
-  toJSON(): string;
 
   /**
-   * Returns the size of this type.
+   * Serialize the {@link Field} to a JSON string, e.g. for printing. If you try to print a {@link Field} without this function it will directly stringify the Field object, resulting in an unreadable output.
+   * 
+   * **Warning**: This operation does NOT affect the circuit and can't be used to prove anything about the JSON string representation of the {@link Field}. Please use it only during debugging.
+   * 
+   * @example
+   * ```ts
+   * const someField = Field(42);
+   * console.log(someField.toJSON());
+   * ```
+   * 
+   * @return A string equivalent to the JSON representation of the {@link Field}.
    */
-  sizeInFields(): number;
+  toJSON(): string;
 
   /**
-   * Serializes this data structure into {@link Field} elements.
+   * This function is the implementation of {@link Provable.toFields} in {@link Field} type.
+   * You can use this array to calculate the {@link Poseidon} hash of a {@link Field}.
+   * This will be always an array of length 1, where the first and only element equals the {@link Field} itself.
+   * 
+   * @return A {@link Field} array of length 1 created from this {@link Field}.
    */
   toFields(): Field[];
 
   /**
+   * Check if this {@link Field} is lower than another "field-like" value.
+   * Returns a {@link Bool}, which is a provable type and can be used prove to the validity of this statement.
    *
-   * Check if this {@link Field} is lower than another Field-like value.
-   * Returns a {@link Bool}.
-   *
+   * @example
    * ```ts
-   * Field(2).lessThan(3); // Bool(true)
+   * Field(2).lessThan(3).assertEquals(Bool(true));
    * ```
-   */
-  lessThan(y: Field | number | string | boolean): Bool;
-  /**
-   *
-   * Check if this {@link Field} is lower than or equal to another Field-like value.
-   * Returns a {@link Bool}.
-   *
+   * 
+   * **Warning**: As this method compares the bigint value of a {@link Field}, it can result in unexpected behaviour when used with negative inputs or modular division.
+   * 
+   * @example
    * ```ts
-   * Field(2).lessThanOrEqual(3); // Bool(true)
+   * Field(1).div(Field(3)).lessThan(Field(1).div(Field(2))).assertEquals(Bool(true)); // This code will throw an error
    * ```
+   * 
+   * @param value - the "field-like" value to compare with this {@link Field}.
+   * 
+   * @return A {@link Bool} representing if this {@link Field} is less than another "field-like" value.
    */
-  lessThanOrEqual(y: Field | number | string | boolean): Bool;
+  lessThan(value: Field | number | string | boolean): Bool;
+
   /**
+   * Check if this {@link Field} is lower than or equal another "field-like" value.
```

Comment:
if we mean less than, same as https://github.com/o1-labs/snarkyjs/pull/917/files#diff-c986331433c1c055aa5d451d0b789e5f5e9aee97f754e6f1550fb09b5acc91b4R360
```suggestion
   * Check if this {@link Field} is less than or equal to another "field-like" value.
```